### PR TITLE
File sharing plugin.

### PR DESCRIPTION
**Pull request by g-tv**
Hi there, browsepy is a nice file sharing solution. I love it.

I wanted to create a plugin that allows me to share files only using a specific key.
The browsepy is behind a proxy and I only need to make the `/share/get/...` path reachable.
Next step would be to add some more features to know if file is shared, also an expiry date.

Looking forward to feedback!

### CODE REVIEW

**Comment by ergoithz** [comment]
Hi @g-tv , 

**TL;DR** This won't be merged as is, but this was the reason of the plugin API redesign. So thanks!

Thank you so much for this pull request, and sorry for the late response, the reason is be explained below.

Sadly this plugin cannot be merged in its current state (see [contributing builtin plugins documentation](http://ergoithz.github.io/browsepy/builtin_plugins.html#contributing-builtin-plugins)).

Said that, **this pull request was definitely useful** because of the following:

1. This plugin made obvious how awful old API was, especially when you need non-mimetype filtering.
2. There was no way to define plugin command line arguments.
3. The absolutely lack of documentation.
4. Typos everywhere.

I hope this issues were fixed now as a new version have been released with the following changes:

1. New plugin API which uses functions for widget filtering.
2. New API for command line arguments.
3. Online documentation is now available at http://ergoithz.github.io/browsepy/ and the reference player plugin is fully functional now
4. Spellchecking, and more coverage.

So, again, thank you so much for your contribution!

**P.S.** I you're still interested on merging your plugin, [as long requirements were met](http://ergoithz.github.io/browsepy/builtin_plugins.html#contributing-builtin-plugins), please fill a new pull request.
